### PR TITLE
DERP-438: Add GLA Core users module for username.

### PR DESCRIPTION
**Pull request by joz-cti**

### CODE REVIEW

**Review by philwolstenholme — APPROVED** [review]

File: modules/gla_core_users/gla_core_users.module [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+function gla_core_users_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Add a custom validate hook onto the user form and user register form.
+  if ($form_id == 'user_form' || $form_id == 'user_register_form') {
+    $form['#validate'][] = 'gla_core_users_username_validate';
+  }
+}
+
+/**
+ * Validates the username based on the configuration.
```
Comment (lukewhitmore) [inline modules/gla_core_users/gla_core_users.module:21]: This is a question for me as much as you, but should we be adding phpdoc params?

File: modules/gla_core_users/gla_core_users.info.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+name: 'GLA Core Users'
+type: module
+description: 'Modify user accounts.'
+core_version_requirement: ^8.8 || ^9
+package: Custom
```
Comment (lukewhitmore) [inline modules/gla_core_users/gla_core_users.info.yml:5]: Should this be `gla_core`?

**Review by lukewhitmore — COMMENTED** [review]
A few questions ..

File: modules/gla_core_users/gla_core_users.info.yml (hunk 2 above)
Comment (joz-cti) [reply modules/gla_core_users/gla_core_users.info.yml:5]: I don't this is a requirement yet for Core packages.

File: modules/gla_core_users/gla_core_users.module (hunk 1 above)
Comment (joz-cti) [reply modules/gla_core_users/gla_core_users.module:21]: I don't think this is required for `hooks_` and validate functions as it's is very easy to figure out what form and form_states are here.

**Pull request merged**